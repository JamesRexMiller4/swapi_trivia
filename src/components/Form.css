@font-face {
  font-family: 'Formal Main';
  src: url('../Fonts/EPISODE1.TTF');
}

@import url('https://fonts.googleapis.com/css?family=Titillium+Web&display=swap');

#form-wrapper {
  background-color: rgba(07, 07, 10, 0.7);
  border: 2px solid #EFF6EE;
  display: block;
  height: 500px;
}

span {
  color: #EEDB00;
  display: block;
  font-size: 1.6rem;
  font-family: 'Formal Main';
  text-decoration: underline;
  word-spacing: 0.4rem;
}

.input-wrapper {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: 25px;
  text-align: left;
  width: 50%;
}

label {
  color: white;
  display: block;
  font-family: 'Titillium Web', sans-serif;
  margin-bottom: 5px;
}

input {
  font-family: 'Titillium Web', sans-serif;
  height: 35px;
  width: 100%;
}

.radio-input {
  width: 20px;
}

#radio-wrapper {
  display: flex;
  justify-content: space-between;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%;
  text-align: center;
}

#button-main-label {
  margin-top: 35px;
}

.radio-label {
  font-size: 0.8rem;
  margin-top: 0;
}

button {
  background-color: #001F54;
  color: white;
  font-family: 'Titillium Web', sans-serif;
  font-size: 1.2rem;
  height: 35px;
  margin-top: 50px;
  outline: none;
  width: 35%;
}

button:active {
  background-color: #EEDB00;
  color: #131319;
}
