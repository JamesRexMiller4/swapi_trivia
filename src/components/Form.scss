@import 'variables.scss';

#form-wrapper {
  background-color: rgba(07, 07, 10, 0.7);
  border: 2px solid #EFF6EE;
  display: block;
  height: 500px;
}

span {
  color: #EEDB00;
  display: block;
  font-size: 1.6rem;
  font-family: $medium-text-font;
  text-decoration: underline;
  word-spacing: 0.4rem;
}

.input-wrapper {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: 25px;
  text-align: left;
  width: 50%;
}

label {
  color: white;
  display: block;
  font-family: $small-text-font;
  margin-bottom: 5px;
}

input {
  font-family: $small-text-font;
  font-size: 1rem;
  height: 35px;
  padding: 5px;
  width: 100%;
}

.radio-input {
  width: 20px;
}

#radio-wrapper {
  display: flex;
  justify-content: space-between;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%;
  text-align: center;
}

#button-main-label {
  margin-top: 35px;
}

.radio-label {
  font-size: 1rem;
  margin-top: 0;
}

button {
  background-color: #001F54;
  color: white;
  font-family: $small-text-font;
  font-size: 1.2rem;
  height: 35px;
  margin-top: 50px;
  outline: none;
  width: 35%;
}

button:active {
  background-color: #EEDB00;
  color: #131319;
}

.error-p {
  font-family: $small-text-font;
  margin-top: 10px;
  margin-bottom: -35px;
  color: red;
}

.hidden {
  display: none;
}
