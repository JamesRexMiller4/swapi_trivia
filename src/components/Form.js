import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      ranking: ''
    }
  }
  render() {
    return (
      <div id="form-wrapper">
        <span>Please sign in to continue.</span>
        <div className="input-wrapper">
          <label>Name:</label>
          <input type="text" id="name-input"></input>
        </div>
        <div className="input-wrapper">
          <label>Favorite Quote:</label>
          <input type="text" id="quote-input"></input>
        </div>
        <div className="input-wrapper">
          <label id="button-main-label">Knowledge Ranking:</label>
          <div id="radio-wrapper">
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Novice" id="novice-input"></input>
              <label className="radio-label" for="novice-input">Novice</label>
            </div>
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Novice" id="intermediate-input"></input>
              <label className="radio-label" for="intermediate-input">Intermediate</label>
            </div>
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Novice" id="expert-input"></input>
              <label className="radio-label" for="expert-input">Expert</label>
            </div>
          </div>
        </div>
        <button id="form-submit">Sign In</button>
      </div>
    )
  }
}

export default Form;
