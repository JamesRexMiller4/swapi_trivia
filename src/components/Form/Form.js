import React, { Component } from 'react';
import './Form.scss';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      ranking: '',
      className: 'hidden'
    }
  }

  handleChange = event => {
    event.target.getAttribute('id') === 'name-input' ?
    this.setState({name: event.target.value})
    : event.target.getAttribute('id') === 'quote-input' ?
    this.setState({quote: event.target.value})
    : event.target.getAttribute('id') === 'novice-input' ?
    this.setState({ranking: 'novice'})
    : event.target.getAttribute('id') === 'intermediate-input' ?
    this.setState({ranking: 'intermediate'})
    : event.target.getAttribute('id') === 'expert-input' ?
    this.setState({ranking: 'expert'}) : console.log(null)
    setTimeout(() => {this.checkErrorStatus()}, 100)
  }

  validateForm = event => {
    (event.target.getAttribute('id') === 'form-submit')
    && this.state.name.length > 0 && this.state.quote.length > 0
    && this.state.ranking.length > 0 ?
    this.props.updateLogin(this.state)
    : this.setState({className: 'error-p'})
  }

  checkErrorStatus = () => {
    this.state.name.length > 0 && this.state.quote.length > 0 && this.state.ranking.length > 0 ?
    this.setState({className: 'hidden'}) : console.log(null);
  }

  render() {
    console.log(this.state);
    return (
      <div id="form-wrapper">
        <span>Please sign in to continue.</span>
        <div className="input-wrapper">
          <label>Name:</label>
          <input type="text" id="name-input" onChange={event => this.handleChange(event)}></input>
        </div>
        <div className="input-wrapper">
          <label>Favorite Quote:</label>
          <input type="text" id="quote-input" onChange={event => this.handleChange(event)}></input>
        </div>
        <div className="input-wrapper">
          <label id="button-main-label">Knowledge Ranking:</label>
          <div id="radio-wrapper">
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Novice" id="novice-input" onClick={event => this.handleChange(event)}></input>
              <label className="radio-label" htmlFor="novice-input">Novice</label>
            </div>
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Intermediate" id="intermediate-input" onClick={event => this.handleChange(event)}></input>
              <label className="radio-label" htmlFor="intermediate-input">Intermediate</label>
            </div>
            <div className="radio-wrapper-sm">
              <input className="radio-input" type="radio" name="Expert" id="expert-input" onClick={event => this.handleChange(event)}></input>
              <label className="radio-label" htmlFor="expert-input">Expert</label>
            </div>
          </div>
        </div>
        <p className={this.state.className}>Please fill out all form fields</p>
        <button id="form-submit" onClick={event => this.validateForm(event)}>Sign In</button>
      </div>
    )
  }
}

export default Form;
