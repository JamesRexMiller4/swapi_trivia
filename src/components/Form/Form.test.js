import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import {shallow} from 'enzyme';

describe('Form', () => {

let wrapper, mockUpdateLogin;

beforeEach(() => {
  mockUpdateLogin = jest.fn()
  wrapper = shallow(<Form updateLogin={mockUpdateLogin}/>)
})

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Form />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = {
      target: {
        name: 'name',
        id: 'name-input',
        value: 'Jerry'}
    };

    const expected = {className: 'hidden', name: 'Jerry', quote: '', ranking: ''};

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state()).toEqual(expected);
  });

  it.skip('should call the method checkErrorStatus when handleChange is called', () => {
    // wrapper.instance().checkErrorStatus = jest.fn()
    const spy = jest.spyOn(wrapper.instance(), 'checkErrorStatus')
    const mockCheckErr = wrapper.instance().checkErrorStatus


    const mockEvent = {
      target: {
        name: 'name',
        id: 'name-input',
        value: 'Jerry'}
    };

    wrapper.instance().handleChange(mockEvent)
  
    expect(spy).toHaveBeenCalled()
    expect(mockCheckErr).toBe(true)
  })

  it('should call the method validateForm when the submit button is clicked', () => {
    wrapper.instance().validateForm = jest.fn();
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.find('#form-submit').simulate('click', mockEvent);

    expect(wrapper.instance().validateForm).toHaveBeenCalledWith(mockEvent)
  });

  it('should call updateLogin when validate form has been called, and all inputs are filled out', () => {
    const userInfo = {name: 'Jerry', quote: "Wasn't me", ranking: 'novice', className: 'hidden'};
    const mockEvent = { preventDefault: jest.fn(), target: {id: 'form-submit'} };
    const expected = {name: 'Jerry', quote: "Wasn't me", ranking: 'novice', className: 'hidden'};

    wrapper.instance().setState(userInfo);
    wrapper.instance().validateForm(mockEvent);

    expect(wrapper.instance().props.updateLogin).toHaveBeenCalledWith(expected)
  })

  it('should update state and display error when validateForm has been called and not all inputs are filled out', () => {
    const defaultState = {name: '', quote: '', ranking: '', className: 'hidden'};
    const expected = {name: '', quote: '', ranking: '', className: 'error-p'};
    const mockEvent = { preventDefault: jest.fn(), target: {id: 'form-submit'} };

    wrapper.instance().setState(defaultState);
    wrapper.instance().validateForm(mockEvent);

    expect(wrapper.state()).toEqual(expected);
  });

  it('should update state.className after displaying error message, as user fills out all inputs', () => {
    const defaultState = {name: '', quote: "Wasn't me", ranking: 'novice', className: 'hidden'};
    const errState = {name: '', quote: "Wasn't me", ranking: 'novice', className: 'error-p'};
    const mockEvent = { preventDefault: jest.fn(), target: {id: 'form-submit'} };
    const validState = {name: 'Jerry', quote: "Wasn't me", ranking: 'novice', className: 'hidden'};
    const mockDetails = {
      target: {
        name: 'name',
        id: 'name-input',
        value: 'Jerry'}
    };

    wrapper.instance().setState(defaultState)
    wrapper.instance().validateForm(mockEvent)
    expect(wrapper.state()).toEqual(errState)
    wrapper.instance().handleChange(mockDetails)
    wrapper.instance().checkErrorStatus()
    expect(wrapper.state()).toEqual(validState)
  });

});