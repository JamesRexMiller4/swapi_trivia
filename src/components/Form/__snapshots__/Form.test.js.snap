// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  autoComplete="off"
  id="form-wrapper"
>
  <span>
    Please sign in to continue.
  </span>
  <div
    className="input-wrapper"
  >
    <label>
      Name:
    </label>
    <input
      id="name-input"
      name="name"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <label>
      Favorite Quote:
    </label>
    <input
      id="quote-input"
      name="quote"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <label
      id="button-main-label"
    >
      Knowledge Ranking:
    </label>
    <div
      id="radio-wrapper"
    >
      <div
        className="radio-wrapper-sm"
      >
        <input
          className="radio-input"
          id="novice-input"
          name="ranking"
          onClick={[Function]}
          type="radio"
        />
        <label
          className="radio-label"
          htmlFor="novice-input"
        >
          Novice
        </label>
      </div>
      <div
        className="radio-wrapper-sm"
      >
        <input
          className="radio-input"
          id="intermediate-input"
          name="ranking"
          onClick={[Function]}
          type="radio"
        />
        <label
          className="radio-label"
          htmlFor="intermediate-input"
        >
          Intermediate
        </label>
      </div>
      <div
        className="radio-wrapper-sm"
      >
        <input
          className="radio-input"
          id="expert-input"
          name="ranking"
          onClick={[Function]}
          type="radio"
        />
        <label
          className="radio-label"
          htmlFor="expert-input"
        >
          Expert
        </label>
      </div>
    </div>
  </div>
  <p
    className="hidden"
  >
    Please fill out all form fields
  </p>
  <button
    id="form-submit"
    onClick={[Function]}
  >
    Sign In
  </button>
</form>
`;
